packages/pn-pa-webapp/src/pages/NewApiKey.page.tsx

```diff
@@ -0,0 +1,185 @@
+import { makeStyles } from '@mui/styles';
+import { useIsMobile, Prompt, PnBreadcrumb } from '@pagopa-pn/pn-commons';
+import { useFormik } from 'formik';
+import * as yup from 'yup';
+import {
+  Box,
+  Typography,
+  TextField,
+  Autocomplete,
+  Paper,
+  Grid,
+  Button,
+  MenuItem,
+  ListItemIcon,
+  ListItemText,
+  Checkbox,
+} from '@mui/material';
+import { useTranslation } from 'react-i18next';
+import * as routes from '../navigation/routes.const';
+import { RootState } from '../redux/store';
+import { useAppDispatch, useAppSelector } from '../redux/hooks';
+import { getApiKeyGroups, saveNewApiKey } from '../redux/NewApiKey/actions';
+import SyncFeedbackApiKey from './components/NewApiKey/SyncFeedbackApiKey';
+
+const useStyles = makeStyles(() => ({
+  root: {
+    '& .paperContainer': {
+      boxShadow: 'none',
+    },
+  },
+}));
+
+const NewApiKey = () => {
+  const dispatch = useAppDispatch();
+  const newApiKey = useAppSelector((state: RootState) => state.newApiKeyState.apiKey);
+  const isMobile = useIsMobile();
+  const groups = useAppSelector((state: RootState) => state.newApiKeyState.groups);
+  const { t } = useTranslation(['apikeys'], { keyPrefix: 'new-api-key' });
+  const { t: tc} = useTranslation(['common']);
+  const [apiKeySent, setApiKeySent] = useState<boolean>(false);
+
+  const initialValues = () => ({
+    name: '',
+    groups: [] as Array<string>,
+  });
+
+  const validationSchema = yup.object({
+    name: yup.string().required(t('form-error-name')),
+    groups: yup.array().min(1, t('form-error-groups')),
+  });
+
+  useEffect(() => {
+    if (groups.length === 0) {
+      void dispatch(getApiKeyGroups());
+    }
+  }, []);
+  
+  const formik = useFormik({
+    initialValues: initialValues(),
+    validateOnMount: true,
+    validationSchema,
+    onSubmit: (values) => {
+      if (formik.isValid) {
+        /*
+          Integrare logica di success / failure ed eventuale callback relativa.
+          Il setState setApiKeySent(true) mostra la schermata di avvenuto successo.
+        */
+
+        void dispatch(saveNewApiKey({ ...values }));
+        setApiKeySent(true);
+      }
+    },
+  });
+
+  const handleChangeTouched = async (e: ChangeEvent) => {
+    formik.handleChange(e);
+    await formik.setFieldTouched(e.target.id, true, false);
+  };
+
+  const classes = useStyles();
+
+  const handleGroupClick = async (_event: any, value: Array<string>) => {
+    await formik.setFieldValue('groups', value);
+    await formik.setFieldTouched('groups', true, false);
+  };
+
+  return (
+    <>
+      {!apiKeySent && (
+        <Prompt
+          title={t('page-title')}
+          message={t('cancel-prompt')}
+          eventTrackingCallbackPromptOpened={() => {}} // impostare eventi tracking previsti
+          eventTrackingCallbackCancel={() => {}} // impostare eventi tracking previsti
+          eventTrackingCallbackConfirm={() => {}} // impostare eventi tracking previsti
+        >
+          <Box p={3}>
```

Comment:
Anche qui usiamo TitleBox come componente in maniera simile a come è stato fatto sulla NewNotification.page

Reply:
@NicolaGiornetta